packages/sdk/src/evm/common/plugin/getCompositePluginABI.ts

```diff
@@ -68,6 +68,22 @@ export async function getCompositeABI(
 
       // get ABIs of extension contracts
       pluginABIs = await getPluginABI(plugins as any[], provider, storage);
+    } else {
+      // check if the contract is diamond pattern
+      const dimaondAbi = [
+        "function facets() external view returns (tuple(address,bytes4[])[])",
```

Comment:
we should be detecting this feature instead of always hitting the RPC to check this. Otherwise you're adding an RPC call to every getContract()

Reply:
The diamond entrypoint doesn't have anything specific besides the fallback function. So tried this way to test it.

However, looking into directly reading storage slot like we do for resolving proxies.